src/finetining/NLLB/finetune.py

```diff
@@ -138,9 +187,21 @@ def evaluate_model(model, tokenizer, val_dataset, max_length):
             y.input_ids[y.input_ids == tokenizer.pad_token_id] = -100
 
             loss = model(**x, labels=y.input_ids).loss
+    
+            # switch the dataset xx and yy to run the training in the other way
+            if multiway:
+                tokenizer.src_lang = lang2
+                x = tokenizer(yy, return_tensors='pt', padding=True, truncation=True, max_length=max_length).to(model.device)
+
+                tokenizer.tgt_lang = lang1
+                y = tokenizer(xx, return_tensors='pt', padding=True, truncation=True, max_length=max_length).to(model.device)
+                y.input_ids[y.input_ids == tokenizer.pad_token_id] = -100
+
+                loss += model(**x, labels=y.input_ids).loss
+                
+
```

Comment:
The input x was passed to the model as seen here `   loss += model(**x, labels=y.input_ids).loss`